### ISSUE TITLE
Shadowlands patch errors

### ISSUE THREAD

**Issue by Mythris29** [wow-addon/FarmIt2#2]
Got these errors today on login.
I'm assuming it had something to do with these changes
https://www.townlong-yak.com/framexml/beta/Blizzard_APIDocumentation#C_CurrencyInfo.GetCurrencyInfo

28x FarmIt2\FarmIt2_Currency.lua:283: attempt to call global 'GetBackpackCurrencyInfo' (a nil value)
[string "@FarmIt2\FarmIt2_Currency.lua"]:283: in function <FarmIt2\FarmIt2_Currency.lua:278>

Locals:
self = FI_Currency_3 {
 0 = <userdata>
}
cid = 3
(*temporary) = nil
(*temporary) = 3
(*temporary) = "attempt to call global 'GetBackpackCurrencyInfo' (a nil value)"

----------------------------------------------------------------------------------------------------------

586x FarmIt2\FarmIt2_Currency.lua:85: attempt to call global 'GetBackpackCurrencyInfo' (a nil value)
[string "@FarmIt2\FarmIt2_Currency.lua"]:85: in function `FI_Update_Currency'
[string "@FarmIt2\FarmIt2.lua"]:136: in function `FI_OnEvent'
[string "*:OnEvent"]:1: in function <[string "*:OnEvent"]:1>

Locals:
event = "CURRENCY_DISPLAY_UPDATE"
temp = <table> {
 1 = <table> {
 }
 2 = <table> {
 }
 3 = <table> {
 }
}
(for index) = 1
(for limit) = 3
(for step) = 1
cid = 1
(*temporary) = nil
(*temporary) = 1
(*temporary) = "attempt to call global 'GetBackpackCurrencyInfo' (a nil value)"

-----------------------------------------------------------------------------------------

29x FarmIt2\FarmIt2_Currency.lua:25: attempt to call global 'GetCurrencyListSize' (a nil value)
[string "@FarmIt2\FarmIt2_Currency.lua"]:25: in function `FI_HasTokens'
[string "@FarmIt2\FarmIt2_Currency.lua"]:49: in function `FI_Init_Currency'
[string "@FarmIt2\FarmIt2_Core.lua"]:129: in function `FI_Load'
[string "@FarmIt2\FarmIt2.lua"]:128: in function `FI_OnEvent'
[string "*:OnEvent"]:1: in function <[string "*:OnEvent"]:1>

Locals:
currency = nil
(*temporary) = 1
(*temporary) = nil
(*temporary) = "attempt to call global 'GetCurrencyListSize' (a nil value)"

**Comment by davidcraig** [comment]
I've just pushed a potential fix, I only using it in a limited fashion so not sure if i've covered everything, if you do get any errors let me know when/how it happens, thanks :)

**Comment by Mythris29** [comment]
So far so good. As always thank you very much for your work on this addon. I was devastated when it wasn't supported anymore so you have at least one person that really appreciates your work on this :)  

I'll let you know if I run into further errors as I get deeper into the game.

**Comment by karpana** [comment]
@davidcraig 

Where is the possible fix been pushed to?
I'm not seeing it show in Twitch addon manager.

**Comment by karpana** [comment]
@davidcraig 

I found the modified file in the code section.  :-)
and I copied/replaced it overtop of the existing file in my addons folder.

Seems to be working okay... but I'm still getting one error when the addon first loads at login.
It shows up 3 times in my LUA error logger screen:

```
Message: Interface\AddOns\FarmIt2\FarmIt2_Currency.lua:183: bad argument #1 to 'strlen' (string expected, got table)
Time: Wed Oct 14 20:37:14 2020
Count: 1
Stack: Interface\AddOns\FarmIt2\FarmIt2_Currency.lua:183: bad argument #1 to 'strlen' (string expected, got table)
[string "=[C]"]: in function `strlen'
[string "@Interface\AddOns\FarmIt2\FarmIt2_Currency.lua"]:183: in function `FI_Update_Currency_Slot'
[string "@Interface\AddOns\FarmIt2\FarmIt2_Currency.lua"]:172: in function `FI_Update_Currency_Bar'
[string "@Interface\AddOns\FarmIt2\FarmIt2_Currency.lua"]:143: in function `FI_Update_Currency'
[string "@Interface\AddOns\FarmIt2\FarmIt2_Currency.lua"]:57: in function `FI_Init_Currency'
[string "@Interface\AddOns\FarmIt2\FarmIt2_Core.lua"]:129: in function `FI_Load'
[string "@Interface\AddOns\FarmIt2\FarmIt2.lua"]:93: in function `FI_OnEvent'
[string "*:OnEvent"]:1: in function <[string "*:OnEvent"]:1>

Locals: (*temporary) = <table> {
 quantity = 1811
 currencyTypesID = 1719
 name = "Corrupted Mementos"
 iconFileID = 646678
}
(*temporary) = "string expected, got table"

```

**Comment by Barleduq** [comment]
TL;DR: CurseBreaker, an updater from github that works on Linux, supposedly can also pull from github. When I tried what I thought was the correct syntax, I got this: "This integration only supports the projects that provide packaged releases." Could you tell me what I should be pulling, or work with the Cursebreaker folks to put things correct?  

Hi there.  I'm an absolute rank newbie in terms of things that aren't packaged neatly as addons.  I've been manually downloading and extracting addons, as I run on Linux and none of the automagic updaters work on linux, but I can't figure out how to get a 'package' from here.  I'm also trying out the updater called CurseBreaker (which I actually got from github), which claims to be able to pull packages from curseforge, wowinterface, Elvui, and github (and I think a few more), but giving it the url of this (which supposedly works) gets the error "This integration supports only the projects that provide packaged releases".

I honestly don't know how to go about making this work. I hope, in giving you this information (and I expect it's too much information), you can somehow connect with Cursebreaker and find out what they need?   I've been using this since at least WoD, and coped with various errors, and would like to keep using it, please.  (Or, do you host it on curseforge? the one there didn't seem to have been updated in forever, and when I lost whatever fixed files I had when a hard disk failure back in February made me reinstall the world, it was back to getting Soundkit errors...)

-Barleduq

**Comment by Mythris29** [comment]
Found some errors today on another alt

1027x FarmIt2\FarmIt2_Currency.lua:183: bad argument #1 to 'strlen' (string expected, got table)
[string "=[C]"]: in function `strlen'
[string "@FarmIt2\FarmIt2_Currency.lua"]:183: in function `FI_Update_Currency_Slot'
[string "@FarmIt2\FarmIt2_Currency.lua"]:172: in function `FI_Update_Currency_Bar'
[string "@FarmIt2\FarmIt2_Currency.lua"]:143: in function `FI_Update_Currency'
[string "@FarmIt2\FarmIt2_Currency.lua"]:57: in function `FI_Init_Currency'
[string "@FarmIt2\FarmIt2_Core.lua"]:129: in function `FI_Load'
[string "@FarmIt2\FarmIt2.lua"]:128: in function `FI_OnEvent'
[string "*:OnEvent"]:1: in function <[string "*:OnEvent"]:1>

Locals:
(*temporary) = <table> {
 currencyTypesID = 515
 quantity = 0
 name = "Darkmoon Prize Ticket"
 iconFileID = 134481
}
(*temporary) = "string expected, got table"

**Comment by davidcraig** [comment]
Thanks for the reports, i've also seen this now, will try and fix as soon as possible.

**Comment by jgkg1999** [comment]
Thanks so much for doing this. This addon is one of my favorites and I was so bummed when they stopped updating it. Can't wait for it to be working again without the errors :)

**Comment by davidcraig** [comment]
I've just pushed another change, hopefully it helps the code around lines ~183/184

```Message: Interface\AddOns\FarmIt2\FarmIt2_Currency.lua:183: bad argument #1 to 'strlen' (string expected, got table)```

**Comment by Mythris29** [comment]
Looks like a new error popped up on one of my alts. As of your last push I'm not seeing the other errors I reported so progress :)

341x FarmIt2\FarmIt2_Core.lua:714: attempt to concatenate field 'name' (a table value)
[string "@FarmIt2\FarmIt2_Core.lua"]:714: in function `FI_Progress'
[string "@FarmIt2\FarmIt2_Currency.lua"]:136: in function `FI_Update_Currency'
[string "@FarmIt2\FarmIt2_Core.lua"]:675: in function `FI_Update'
[string "@FarmIt2\FarmIt2.lua"]:133: in function `FI_OnEvent'
[string "*:OnEvent"]:1: in function <[string "*:OnEvent"]:1>

Locals:
data = <table> {
 success = false
 name = <table> {
 }
 objective = 0
 id = 1
 lastcount = 0
 icon = 134481
 count = 0
}
silent = nil
f_name = "FI_Currency_1"
database = "Currencies"
currencyName = nil
status = nil
info = nil
suffix = nil
color = nil
sound = nil
itemName = nil
itemLink = nil
(*temporary) = "|cFF"
(*temporary) = "3399FF"
(*temporary) = <table> {
 currencyTypesID = 515
 quantity = 0
 name = "Darkmoon Prize Ticket"
 iconFileID = 134481
}
(*temporary) = "|r"
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = "attempt to concatenate field 'name' (a table value)"

**Comment by jgkg1999** [comment]
This is what I'm seeing now:

 Interface\AddOns\FarmIt2\FarmIt2_Currency.lua:25: attempt to index global 'CurrencyInfo' (a nil value)
Time: Thu Oct 22 11:35:26 2020
Count: 1
Stack: Interface\AddOns\FarmIt2\FarmIt2_Currency.lua:25: attempt to index global 'CurrencyInfo' (a nil value)
[string "@Interface\AddOns\FarmIt2\FarmIt2_Currency.lua"]:25: in function `FI_HasTokens'
[string "@Interface\AddOns\FarmIt2\FarmIt2_Currency.lua"]:48: in function `FI_Init_Currency'
[string "@Interface\AddOns\FarmIt2\FarmIt2_Core.lua"]:129: in function `FI_Load'
[string "@Interface\AddOns\FarmIt2\FarmIt2.lua"]:128: in function `FI_OnEvent'
[string "*:OnEvent"]:1: in function <[string "*:OnEvent"]:1>

Locals: currency = nil
(*temporary) = 1
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = "attempt to index global 'CurrencyInfo' (a nil value)"

**Comment by davidcraig** [comment]
A new update has been pushed, please let me know if you still experience problems with the new version.

**Comment by Mythris29** [comment]
So far so good!
